ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -54,63 +59,29 @@ class SurveyRepository
 constructor(
   private val localDataStore: LocalDataStore,
   private val remoteDataStore: RemoteDataStore,
-  private val localValueStore: LocalValueStore
+  private val localValueStore: LocalValueStore,
+  @ApplicationScope private val externalScope: CoroutineScope,
```

Comment:
Here too, I personally find the use of injection more confusing and less readable than just using the platform and standard library directly.

Here, I need to recall what `externalScope` means in terms of CoroutineScopes, contrarily, I think just using Android's scopes directly is a lot clearer (https://developer.android.com/topic/libraries/architecture/coroutines#lifecyclescope)

Reply:
Same comment as above. Relevant note from the best practices guide:
<img width="959" alt="image" src="https://user-images.githubusercontent.com/228050/213500417-f5f89ce0-24aa-4e48-aa68-444db3d27c6f.png">

Reply:
Yep, saw it. Like that they have more examples of the non-best practice in their docs and then have single, easy to miss note explaining that they are in fact not recommended practices 🙃